FrontEnd/src/BookingPage/BookingDB.js

```diff
@@ -0,0 +1,108 @@
+import { database } from "../UserAuthentication/FirebaseConfig";
+import {
+  query,
+  where,
+  collection,
+  doc,
+  getDocs,
+  setDoc,
+  getDoc,
+  onSnapshot,
+} from "firebase/firestore";
+
+const DATABASE_FOLDER_NAME = "users";
+const requestAppointment = async (
+  userUID,
+  vendorUID,
+  appointmentData,
+  userData
+) => {
+  try {
+    const Status = "pending";
```

Comment:
```suggestion
    const PENDING_STATUS = "pending";
```

Reply:
The Status is subject to chnage based on user's action. It can be modified to "accepted" or "declined"